File: backend/src/models/session.ts
import mongoose from "mongoose";

const sessionSchema = new mongoose.Schema({
    movie: {
        type: mongoose.Schema.Types.ObjectId,
        ref: "Movie",
        required: true,
    },
    hall:{
        type: mongoose.Schema.Types.ObjectId,
        ref: "Hall",
        required: true,
    },
    price: {
        type: Number,
        required: true,
    },
    seats: {
        type: [[Boolean]],
        required: true
    },
    start_time: {
        type: Date,
        required: true,
    },
    end_time:{
        type: Date,
        required: true
    }
})
const Session = mongoose.model("Session", sessionSchema);
export default Session;

File: FilmHub/Model/Session.swift
//
//  Session.swift
//  FilmHub
//
//  Created by Zakhar Litvinchuk on 20.05.2024.
//

import Foundation

struct Session: Codable {
    let id: String
    let movie: Movie
    let hall: Hall
    let seats: [[Bool]]
    let startTime: Date
    let endTime: Date
    let price: Double
    
    var formattedStartTime: String {
        let formatter = DateFormatter()
        formatter.dateFormat = "HH:mm"
        return formatter.string(from: startTime)
    }
    
    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case movie
        case hall
        case seats
        case startTime = "start_time"
        case endTime = "end_time"
        case price
    }
    
}
